File: backend/src/main/java/com/techgarage/service/AuthService.java
package com.techgarage.service;

import com.techgarage.dto.auth.AuthResponse;
import com.techgarage.dto.auth.LoginRequest;
import com.techgarage.dto.auth.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    void verifyEmail(String token);
    void requestPasswordReset(String email);
    void resendVerification(String email);
    void resetPassword(String token, String newPassword);
}

File: frontend/src/services/authService.js
import api, { unwrap } from './api'

export const authService = {
  register: (payload) => unwrap(api.post('/auth/register', payload)),
  login: (payload) => unwrap(api.post('/auth/login', payload)),
  verifyEmail: (token) => unwrap(api.get('/auth/verify-email', { params: { token } })),
  resendVerification: (payload) => unwrap(api.post('/auth/resend-verification', payload)),
  forgotPassword: (payload) => unwrap(api.post('/auth/forgot-password', payload)),
  resetPassword: (payload) => unwrap(api.post('/auth/reset-password', payload)),
}
